File: Factorial.cpp
#include<iostream>
using namespace std;

class Factorial
{
    public:
      int No;

      Factorial(int iValue)
      {
          No = iValue;
      }

      int Fact()
      {
          int iCnt = 0;
          int iFact = 1;

          for(iCnt = 1; iCnt <= No; iCnt++)
          {
              iFact = iFact * iCnt;
          }
          return iFact;
      }

};

int main()
{
    int iNo = 0;
    int iRet = 0;

    cout<<"Enter the number to find factorial :\n";
    cin>>iNo;

    Factorial fobj(iNo);

    iRet = fobj.Fact();

    cout<<"Factorial of "<<iNo<<" is "<<iRet<<"\n";

    return 0;
}

File: Factorial.java
import java.util.*;

class Mathematic
{
    public int Fact(int No)
    {
        int iCnt = 0;
        int iFact = 1;

        for(iCnt = 1; iCnt <= No; iCnt++)
        {
            iFact = iFact * iCnt;
        }
        return iFact;
    }

}

class Factorial
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        int iNo = 0;
        int iRet = 0;

        System.out.println("Enter the number to find factorial: ");
        iNo = sobj.nextInt();

        Mathematic mobj = new Mathematic();

        iRet = mobj.Fact(iNo);

        System.out.println("Factorial of "+iNo +" is " +iRet);
    }
}